test/snapshot.spec.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+
+const t = require('tap')
+const { readFileSync } = require('fs')
+const { file: tmpFile } = require('tmp-promise')
+
+const generateHeapSnapshot = require('../src/snapshot')
+const validate = require('./fixtures/snapshotSchema')
+
+t.test('it correctly generates a snapshot using promises', async t => {
+  const { path: destination, cleanup } = await tmpFile()
+
+  await generateHeapSnapshot({ destination })
+
+  const generated = JSON.parse(readFileSync(destination, 'utf-8'))
+
+  t.true(validate(generated))
+
+  cleanup()
+})
+
+t.test('it correctly generates a snapshot using callbacks', t => {
+  tmpFile().then(({ path: destination, cleanup }) => {
```

Comment:
please use an async function and await instead.

Reply:
Done!